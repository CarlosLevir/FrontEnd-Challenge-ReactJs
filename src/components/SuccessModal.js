import React, { Component } from 'react';
import { Button, Modal } from 'react-bootstrap';

export default class SuccessModal extends Component {
  render() {
    const { show, message, handleHide } = this.props;
    return (
      <div>
        <div className="modal-container" style={{ height: 200 }}>
          <Modal
            show={show}
            onHide={this.handleHide}
            container={this}
            aria-labelledby="contained-modal-title"
          >
            <Modal.Header closeButton>
              <Modal.Title id="contained-modal-title">Parab√©ns!!!</Modal.Title>
            </Modal.Header>
            <Modal.Body>{message}</Modal.Body>
            <Modal.Footer>
              <Button onClick={handleHide}>Close</Button>
            </Modal.Footer>
          </Modal>
        </div>
      </div>
    );
  }
}
