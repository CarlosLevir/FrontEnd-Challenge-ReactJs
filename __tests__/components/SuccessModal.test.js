import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import SuccessModal from '../../src/components/SuccessModal';

configure({ adapter: new Adapter() });

describe('SuccessModal', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SuccessModal />);
    expect(wrapper).toMatchSnapshot();
  });
});
