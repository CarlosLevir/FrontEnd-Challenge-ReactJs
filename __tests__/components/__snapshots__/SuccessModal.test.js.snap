// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessModal renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SuccessModal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="modal-container"
        style={
          Object {
            "height": 200,
          }
        }
      >
        <Modal
          animation={true}
          aria-labelledby="contained-modal-title"
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          container={
            SuccessModal {
              "context": Object {},
              "props": Object {},
              "refs": Object {},
              "setState": [Function],
              "state": null,
              "updater": Updater {
                "_callbacks": Array [],
                "_renderer": ReactShallowRenderer {
                  "_context": Object {},
                  "_element": <SuccessModal />,
                  "_forcedUpdate": false,
                  "_instance": [Circular],
                  "_newState": null,
                  "_rendered": <div>
                    <div
                      className="modal-container"
                      style={
                        Object {
                          "height": 200,
                        }
                      }
                    >
                      <Modal
                        animation={true}
                        aria-labelledby="contained-modal-title"
                        autoFocus={true}
                        backdrop={true}
                        bsClass="modal"
                        container={[Circular]}
                        dialogComponentClass={[Function]}
                        enforceFocus={true}
                        keyboard={true}
                        manager={
                          ModalManager {
                            "add": [Function],
                            "containers": Array [],
                            "data": Array [],
                            "handleContainerOverflow": true,
                            "hideSiblingNodes": true,
                            "isTopModal": [Function],
                            "modals": Array [],
                            "remove": [Function],
                          }
                        }
                        onHide={[Function]}
                        renderBackdrop={[Function]}
                        restoreFocus={true}
                        show={false}
                      >
                        <ModalHeader
                          bsClass="modal-header"
                          closeButton={true}
                          closeLabel="Close"
                        >
                          <ModalTitle
                            bsClass="modal-title"
                            componentClass="h4"
                            id="contained-modal-title"
                          >
                            Parabéns!!!
                          </ModalTitle>
                        </ModalHeader>
                        <ModalBody
                          bsClass="modal-body"
                          componentClass="div"
                        />
                        <ModalFooter
                          bsClass="modal-footer"
                          componentClass="div"
                        >
                          <Button
                            active={false}
                            block={false}
                            bsClass="btn"
                            bsStyle="default"
                            disabled={false}
                          >
                            Close
                          </Button>
                        </ModalFooter>
                      </Modal>
                    </div>
                  </div>,
                  "_rendering": false,
                  "_updater": [Circular],
                },
              },
              Symbol(enzyme.__setState__): [Function],
            }
          }
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
        >
          <ModalHeader
            bsClass="modal-header"
            closeButton={true}
            closeLabel="Close"
          >
            <ModalTitle
              bsClass="modal-title"
              componentClass="h4"
              id="contained-modal-title"
            >
              Parabéns!!!
            </ModalTitle>
          </ModalHeader>
          <ModalBody
            bsClass="modal-body"
            componentClass="div"
          />
          <ModalFooter
            bsClass="modal-footer"
            componentClass="div"
          >
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
            >
              Close
            </Button>
          </ModalFooter>
        </Modal>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Modal
          animation={true}
          aria-labelledby="contained-modal-title"
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          container={
            SuccessModal {
              "context": Object {},
              "props": Object {},
              "refs": Object {},
              "setState": [Function],
              "state": null,
              "updater": Updater {
                "_callbacks": Array [],
                "_renderer": ReactShallowRenderer {
                  "_context": Object {},
                  "_element": <SuccessModal />,
                  "_forcedUpdate": false,
                  "_instance": [Circular],
                  "_newState": null,
                  "_rendered": <div>
                    <div
                      className="modal-container"
                      style={
                        Object {
                          "height": 200,
                        }
                      }
                    >
                      <Modal
                        animation={true}
                        aria-labelledby="contained-modal-title"
                        autoFocus={true}
                        backdrop={true}
                        bsClass="modal"
                        container={[Circular]}
                        dialogComponentClass={[Function]}
                        enforceFocus={true}
                        keyboard={true}
                        manager={
                          ModalManager {
                            "add": [Function],
                            "containers": Array [],
                            "data": Array [],
                            "handleContainerOverflow": true,
                            "hideSiblingNodes": true,
                            "isTopModal": [Function],
                            "modals": Array [],
                            "remove": [Function],
                          }
                        }
                        onHide={[Function]}
                        renderBackdrop={[Function]}
                        restoreFocus={true}
                        show={false}
                      >
                        <ModalHeader
                          bsClass="modal-header"
                          closeButton={true}
                          closeLabel="Close"
                        >
                          <ModalTitle
                            bsClass="modal-title"
                            componentClass="h4"
                            id="contained-modal-title"
                          >
                            Parabéns!!!
                          </ModalTitle>
                        </ModalHeader>
                        <ModalBody
                          bsClass="modal-body"
                          componentClass="div"
                        />
                        <ModalFooter
                          bsClass="modal-footer"
                          componentClass="div"
                        >
                          <Button
                            active={false}
                            block={false}
                            bsClass="btn"
                            bsStyle="default"
                            disabled={false}
                          >
                            Close
                          </Button>
                        </ModalFooter>
                      </Modal>
                    </div>
                  </div>,
                  "_rendering": false,
                  "_updater": [Circular],
                },
              },
              Symbol(enzyme.__setState__): [Function],
            }
          }
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
        >
          <ModalHeader
            bsClass="modal-header"
            closeButton={true}
            closeLabel="Close"
          >
            <ModalTitle
              bsClass="modal-title"
              componentClass="h4"
              id="contained-modal-title"
            >
              Parabéns!!!
            </ModalTitle>
          </ModalHeader>
          <ModalBody
            bsClass="modal-body"
            componentClass="div"
          />
          <ModalFooter
            bsClass="modal-footer"
            componentClass="div"
          >
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
            >
              Close
            </Button>
          </ModalFooter>
        </Modal>,
        "className": "modal-container",
        "style": Object {
          "height": 200,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animation": true,
          "aria-labelledby": "contained-modal-title",
          "autoFocus": true,
          "backdrop": true,
          "bsClass": "modal",
          "children": Array [
            <ModalHeader
              bsClass="modal-header"
              closeButton={true}
              closeLabel="Close"
            >
              <ModalTitle
                bsClass="modal-title"
                componentClass="h4"
                id="contained-modal-title"
              >
                Parabéns!!!
              </ModalTitle>
            </ModalHeader>,
            <ModalBody
              bsClass="modal-body"
              componentClass="div"
            />,
            <ModalFooter
              bsClass="modal-footer"
              componentClass="div"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
              >
                Close
              </Button>
            </ModalFooter>,
          ],
          "container": SuccessModal {
            "context": Object {},
            "props": Object {},
            "refs": Object {},
            "setState": [Function],
            "state": null,
            "updater": Updater {
              "_callbacks": Array [],
              "_renderer": ReactShallowRenderer {
                "_context": Object {},
                "_element": <SuccessModal />,
                "_forcedUpdate": false,
                "_instance": [Circular],
                "_newState": null,
                "_rendered": <div>
                  <div
                    className="modal-container"
                    style={
                      Object {
                        "height": 200,
                      }
                    }
                  >
                    <Modal
                      animation={true}
                      aria-labelledby="contained-modal-title"
                      autoFocus={true}
                      backdrop={true}
                      bsClass="modal"
                      container={[Circular]}
                      dialogComponentClass={[Function]}
                      enforceFocus={true}
                      keyboard={true}
                      manager={
                        ModalManager {
                          "add": [Function],
                          "containers": Array [],
                          "data": Array [],
                          "handleContainerOverflow": true,
                          "hideSiblingNodes": true,
                          "isTopModal": [Function],
                          "modals": Array [],
                          "remove": [Function],
                        }
                      }
                      onHide={[Function]}
                      renderBackdrop={[Function]}
                      restoreFocus={true}
                      show={false}
                    >
                      <ModalHeader
                        bsClass="modal-header"
                        closeButton={true}
                        closeLabel="Close"
                      >
                        <ModalTitle
                          bsClass="modal-title"
                          componentClass="h4"
                          id="contained-modal-title"
                        >
                          Parabéns!!!
                        </ModalTitle>
                      </ModalHeader>
                      <ModalBody
                        bsClass="modal-body"
                        componentClass="div"
                      />
                      <ModalFooter
                        bsClass="modal-footer"
                        componentClass="div"
                      >
                        <Button
                          active={false}
                          block={false}
                          bsClass="btn"
                          bsStyle="default"
                          disabled={false}
                        >
                          Close
                        </Button>
                      </ModalFooter>
                    </Modal>
                  </div>
                </div>,
                "_rendering": false,
                "_updater": [Circular],
              },
            },
            Symbol(enzyme.__setState__): [Function],
          },
          "dialogComponentClass": [Function],
          "enforceFocus": true,
          "keyboard": true,
          "manager": ModalManager {
            "add": [Function],
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "isTopModal": [Function],
            "modals": Array [],
            "remove": [Function],
          },
          "onHide": [Function],
          "renderBackdrop": [Function],
          "restoreFocus": true,
          "show": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-header",
              "children": <ModalTitle
                bsClass="modal-title"
                componentClass="h4"
                id="contained-modal-title"
              >
                Parabéns!!!
              </ModalTitle>,
              "closeButton": true,
              "closeLabel": "Close",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "modal-title",
                "children": "Parabéns!!!",
                "componentClass": "h4",
                "id": "contained-modal-title",
              },
              "ref": null,
              "rendered": "Parabéns!!!",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-body",
              "children": undefined,
              "componentClass": "div",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-footer",
              "children": <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
              >
                Close
              </Button>,
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsStyle": "default",
                "children": "Close",
                "disabled": false,
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "Close",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="modal-container"
          style={
            Object {
              "height": 200,
            }
          }
        >
          <Modal
            animation={true}
            aria-labelledby="contained-modal-title"
            autoFocus={true}
            backdrop={true}
            bsClass="modal"
            container={
              SuccessModal {
                "context": Object {},
                "props": Object {},
                "refs": Object {},
                "setState": [Function],
                "state": null,
                "updater": Updater {
                  "_callbacks": Array [],
                  "_renderer": ReactShallowRenderer {
                    "_context": Object {},
                    "_element": <SuccessModal />,
                    "_forcedUpdate": false,
                    "_instance": [Circular],
                    "_newState": null,
                    "_rendered": <div>
                      <div
                        className="modal-container"
                        style={
                          Object {
                            "height": 200,
                          }
                        }
                      >
                        <Modal
                          animation={true}
                          aria-labelledby="contained-modal-title"
                          autoFocus={true}
                          backdrop={true}
                          bsClass="modal"
                          container={[Circular]}
                          dialogComponentClass={[Function]}
                          enforceFocus={true}
                          keyboard={true}
                          manager={
                            ModalManager {
                              "add": [Function],
                              "containers": Array [],
                              "data": Array [],
                              "handleContainerOverflow": true,
                              "hideSiblingNodes": true,
                              "isTopModal": [Function],
                              "modals": Array [],
                              "remove": [Function],
                            }
                          }
                          onHide={[Function]}
                          renderBackdrop={[Function]}
                          restoreFocus={true}
                          show={false}
                        >
                          <ModalHeader
                            bsClass="modal-header"
                            closeButton={true}
                            closeLabel="Close"
                          >
                            <ModalTitle
                              bsClass="modal-title"
                              componentClass="h4"
                              id="contained-modal-title"
                            >
                              Parabéns!!!
                            </ModalTitle>
                          </ModalHeader>
                          <ModalBody
                            bsClass="modal-body"
                            componentClass="div"
                          />
                          <ModalFooter
                            bsClass="modal-footer"
                            componentClass="div"
                          >
                            <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              disabled={false}
                            >
                              Close
                            </Button>
                          </ModalFooter>
                        </Modal>
                      </div>
                    </div>,
                    "_rendering": false,
                    "_updater": [Circular],
                  },
                },
                Symbol(enzyme.__setState__): [Function],
              }
            }
            dialogComponentClass={[Function]}
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
          >
            <ModalHeader
              bsClass="modal-header"
              closeButton={true}
              closeLabel="Close"
            >
              <ModalTitle
                bsClass="modal-title"
                componentClass="h4"
                id="contained-modal-title"
              >
                Parabéns!!!
              </ModalTitle>
            </ModalHeader>
            <ModalBody
              bsClass="modal-body"
              componentClass="div"
            />
            <ModalFooter
              bsClass="modal-footer"
              componentClass="div"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
              >
                Close
              </Button>
            </ModalFooter>
          </Modal>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Modal
            animation={true}
            aria-labelledby="contained-modal-title"
            autoFocus={true}
            backdrop={true}
            bsClass="modal"
            container={
              SuccessModal {
                "context": Object {},
                "props": Object {},
                "refs": Object {},
                "setState": [Function],
                "state": null,
                "updater": Updater {
                  "_callbacks": Array [],
                  "_renderer": ReactShallowRenderer {
                    "_context": Object {},
                    "_element": <SuccessModal />,
                    "_forcedUpdate": false,
                    "_instance": [Circular],
                    "_newState": null,
                    "_rendered": <div>
                      <div
                        className="modal-container"
                        style={
                          Object {
                            "height": 200,
                          }
                        }
                      >
                        <Modal
                          animation={true}
                          aria-labelledby="contained-modal-title"
                          autoFocus={true}
                          backdrop={true}
                          bsClass="modal"
                          container={[Circular]}
                          dialogComponentClass={[Function]}
                          enforceFocus={true}
                          keyboard={true}
                          manager={
                            ModalManager {
                              "add": [Function],
                              "containers": Array [],
                              "data": Array [],
                              "handleContainerOverflow": true,
                              "hideSiblingNodes": true,
                              "isTopModal": [Function],
                              "modals": Array [],
                              "remove": [Function],
                            }
                          }
                          onHide={[Function]}
                          renderBackdrop={[Function]}
                          restoreFocus={true}
                          show={false}
                        >
                          <ModalHeader
                            bsClass="modal-header"
                            closeButton={true}
                            closeLabel="Close"
                          >
                            <ModalTitle
                              bsClass="modal-title"
                              componentClass="h4"
                              id="contained-modal-title"
                            >
                              Parabéns!!!
                            </ModalTitle>
                          </ModalHeader>
                          <ModalBody
                            bsClass="modal-body"
                            componentClass="div"
                          />
                          <ModalFooter
                            bsClass="modal-footer"
                            componentClass="div"
                          >
                            <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              disabled={false}
                            >
                              Close
                            </Button>
                          </ModalFooter>
                        </Modal>
                      </div>
                    </div>,
                    "_rendering": false,
                    "_updater": [Circular],
                  },
                },
                Symbol(enzyme.__setState__): [Function],
              }
            }
            dialogComponentClass={[Function]}
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
          >
            <ModalHeader
              bsClass="modal-header"
              closeButton={true}
              closeLabel="Close"
            >
              <ModalTitle
                bsClass="modal-title"
                componentClass="h4"
                id="contained-modal-title"
              >
                Parabéns!!!
              </ModalTitle>
            </ModalHeader>
            <ModalBody
              bsClass="modal-body"
              componentClass="div"
            />
            <ModalFooter
              bsClass="modal-footer"
              componentClass="div"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
              >
                Close
              </Button>
            </ModalFooter>
          </Modal>,
          "className": "modal-container",
          "style": Object {
            "height": 200,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animation": true,
            "aria-labelledby": "contained-modal-title",
            "autoFocus": true,
            "backdrop": true,
            "bsClass": "modal",
            "children": Array [
              <ModalHeader
                bsClass="modal-header"
                closeButton={true}
                closeLabel="Close"
              >
                <ModalTitle
                  bsClass="modal-title"
                  componentClass="h4"
                  id="contained-modal-title"
                >
                  Parabéns!!!
                </ModalTitle>
              </ModalHeader>,
              <ModalBody
                bsClass="modal-body"
                componentClass="div"
              />,
              <ModalFooter
                bsClass="modal-footer"
                componentClass="div"
              >
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                >
                  Close
                </Button>
              </ModalFooter>,
            ],
            "container": SuccessModal {
              "context": Object {},
              "props": Object {},
              "refs": Object {},
              "setState": [Function],
              "state": null,
              "updater": Updater {
                "_callbacks": Array [],
                "_renderer": ReactShallowRenderer {
                  "_context": Object {},
                  "_element": <SuccessModal />,
                  "_forcedUpdate": false,
                  "_instance": [Circular],
                  "_newState": null,
                  "_rendered": <div>
                    <div
                      className="modal-container"
                      style={
                        Object {
                          "height": 200,
                        }
                      }
                    >
                      <Modal
                        animation={true}
                        aria-labelledby="contained-modal-title"
                        autoFocus={true}
                        backdrop={true}
                        bsClass="modal"
                        container={[Circular]}
                        dialogComponentClass={[Function]}
                        enforceFocus={true}
                        keyboard={true}
                        manager={
                          ModalManager {
                            "add": [Function],
                            "containers": Array [],
                            "data": Array [],
                            "handleContainerOverflow": true,
                            "hideSiblingNodes": true,
                            "isTopModal": [Function],
                            "modals": Array [],
                            "remove": [Function],
                          }
                        }
                        onHide={[Function]}
                        renderBackdrop={[Function]}
                        restoreFocus={true}
                        show={false}
                      >
                        <ModalHeader
                          bsClass="modal-header"
                          closeButton={true}
                          closeLabel="Close"
                        >
                          <ModalTitle
                            bsClass="modal-title"
                            componentClass="h4"
                            id="contained-modal-title"
                          >
                            Parabéns!!!
                          </ModalTitle>
                        </ModalHeader>
                        <ModalBody
                          bsClass="modal-body"
                          componentClass="div"
                        />
                        <ModalFooter
                          bsClass="modal-footer"
                          componentClass="div"
                        >
                          <Button
                            active={false}
                            block={false}
                            bsClass="btn"
                            bsStyle="default"
                            disabled={false}
                          >
                            Close
                          </Button>
                        </ModalFooter>
                      </Modal>
                    </div>
                  </div>,
                  "_rendering": false,
                  "_updater": [Circular],
                },
              },
              Symbol(enzyme.__setState__): [Function],
            },
            "dialogComponentClass": [Function],
            "enforceFocus": true,
            "keyboard": true,
            "manager": ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            },
            "onHide": [Function],
            "renderBackdrop": [Function],
            "restoreFocus": true,
            "show": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "modal-header",
                "children": <ModalTitle
                  bsClass="modal-title"
                  componentClass="h4"
                  id="contained-modal-title"
                >
                  Parabéns!!!
                </ModalTitle>,
                "closeButton": true,
                "closeLabel": "Close",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "modal-title",
                  "children": "Parabéns!!!",
                  "componentClass": "h4",
                  "id": "contained-modal-title",
                },
                "ref": null,
                "rendered": "Parabéns!!!",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "modal-body",
                "children": undefined,
                "componentClass": "div",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "modal-footer",
                "children": <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                >
                  Close
                </Button>,
                "componentClass": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": false,
                  "bsClass": "btn",
                  "bsStyle": "default",
                  "children": "Close",
                  "disabled": false,
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": "Close",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
